packages/flame_bloc/lib/src/flame_bloc_reader.dart

```diff
@@ -0,0 +1,30 @@
+import 'dart:async';
+
+import 'package:bloc/bloc.dart';
+import 'package:flame/components.dart';
+import 'package:flutter/material.dart';
+
+import '../flame_bloc.dart';
+
+/// Adds [Bloc] access to a [Component]
+/// Useful for components that needs to only read
+/// a bloc current state or to trigger an event on it
+mixin FlameBlocReader<B extends BlocBase<S>, S> on Component {
+  late B _bloc;
+
+  /// Returns the bloc that this component is reading from
+  B get bloc => _bloc;
+
+  @override
+  @mustCallSuper
+  Future<void> onLoad() async {
+    final providers = ancestors().whereType<FlameBlocProvider<B, S>>();
+    assert(
+      providers.isNotEmpty,
+      'No FlameBlocProvider<$B, $S> available on the component tree',
+    );
+
+    final provider = providers.first;
+    _bloc = provider.bloc;
```

Comment:
This is a great area where a `findeAncestorOfType`-like method on flame would come in handy.

Reply:
Nothing to do here, looks great btw